src/langchain_google_spanner/loader.py

```diff
@@ -0,0 +1,422 @@
+            instance_id: The Spanner instance to load data to.
+            database_id: The Spanner database to load data to.
+            table_name: The table name to load data to.
+            content_column: The name of the content column. Defaulted to the first column.
+            metadata_columns: This is for user to opt-in a selection of columns to use. Defaulted to use
+                              all columns.
+            metadata_json_column: The name of the special JSON column. Defaulted to use "langchain_metadata".
+            client: The connection object to use. This can be used to customized project id and credentials.
+        """
+        self.instance_id = instance_id
+        self.database_id = database_id
+        self.table_name = table_name
+        self.content_column = content_column
+        self.metadata_columns = metadata_columns
+        self.metadata_json_column = metadata_json_column
+        self.client = client_with_user_agent(client, USER_AGENT_LOADER)
+        instance = self.client.instance(instance_id)
+        if not instance.exists():
+            raise Exception("Instance doesn't exist.")
+        database = instance.database(database_id)
+        if not database.exists():
+            raise Exception("Database doesn't exist.")
+        database.reload()
+        self.dialect = database.database_dialect
+        table = database.table(table_name)
+        if not table.exists():
+            raise Exception(
+                "Table doesn't exist. Create table with SpannerDocumentSaver.init_document_table function."
+            )
+        self._table_fields = [
+            n.name for n in table.schema if n.name != metadata_json_column
+        ]
+        self._table_fields.append(metadata_json_column)
+
+    def add_documents(self, documents: List[Document]):
+        """Add documents to the Spanner table."""
+        db = self.client.instance(self.instance_id).database(self.database_id)
+        values = [
+            _load_doc_to_row(
+                self._table_fields, doc, self.content_column, self.metadata_json_column
+            )
+            for doc in documents
+        ]
+
+        for values_batch in _batch(values, MUTATION_BATCH_SIZE):
+            with db.batch() as batch:
+                batch.insert(
+                    table=self.table_name,
+                    columns=self._table_fields,
+                    values=values_batch,
+                )
+
+    def delete(self, documents: List[Document]):
+        """Delete documents from the table."""
+        database = self.client.instance(self.instance_id).database(self.database_id)
+        # load documents to row
+        docs = [
+            _load_doc_to_row(
+                self._table_fields,
+                doc,
+                self.content_column,
+                self.metadata_json_column,
+                False,
+            )
+            for doc in documents
+        ]
+        keys = [[doc[0]] for doc in docs]
```

Comment:
How are we doing this? Are we assuming first field in the tuple would always be id column? if yes how are we ensuring it?